modules/core/src/main/java/org/apache/ignite/dump/DumpReader.java

```diff
@@ -135,13 +151,51 @@ public DumpReader(DumpReaderConfiguration cfg, IgniteLogger log) {
                                     return;
                                 }
 
-                                try (DumpedPartitionIterator iter = dump.iterator(node, grp, part)) {
+                                try (DumpedPartitionIterator iter = new DumpedPartitionIterator() {
+                                    /** */
+                                    final DumpedPartitionIterator delegate = dump.iterator(node, grp, part);
+
+                                    /** */
+                                    final AtomicBoolean consumerProcessingEntry = new AtomicBoolean(false);
```

Comment:
Can we count on `next()` only?

Reply:
No. We can increment recordsProcessed only when consumer finishes processing record, after which hasNext() executed. If we switch only to using next() then we can detect processing of previous record, which means that we can't detect processing of last record, for which normally hasNext() is executed.

Reply:
OK, you can initialize `recordsProcessed` with `-1` and increment it on every `hasNext()`. Does not it look simpler?